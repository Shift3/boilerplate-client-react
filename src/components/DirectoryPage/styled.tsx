import styled from 'styled-components';

export const DirectoryPageContainer = styled.div`
  min-height: ${(props) => props.theme.minPageHeight};
`;