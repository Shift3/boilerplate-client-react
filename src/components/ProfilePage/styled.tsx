import styled from 'styled-components';

export const ProfilePageContainer = styled.div`
  min-height: ${(props) => props.theme.minPageHeight};
`;