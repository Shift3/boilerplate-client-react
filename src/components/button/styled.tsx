import styled from 'styled-components';
import { Button } from 'react-bootstrap';
import StyleValues from '../../utils/styleValues';

const { accent, accentDisabled } = StyleValues;

export const CustomButton = styled(Button)`
  background-color: ${(props) => props.theme.backgroundColor};
  border-color: ${(props) => props.theme.borderColor};
  color: ${(props) => props.theme.color};
  width: ${(props) => props.theme.width};
  &:hover {
    background-color: ${(props) => props.theme.hoverBackgroundColor};
    border-color: ${(props) => props.theme.hoverBorderColor};
  }
  &:disabled {
    background-color: ${(props) => props.theme.disabledBackgroundColor};
    border-color: ${(props) => props.theme.disabledBorderColor};
  }
`;

CustomButton.defaultProps = {
  theme: {
    backgroundColor: accent,
    borderColor: accent,
    color: '#fff',
    disabledBackgroundColor: accentDisabled,
    disabledBorderColor: accent,
    hoverBackgroundColor: accent,
    hoverBorderColor: accent,
    width: '90%'
  }
};