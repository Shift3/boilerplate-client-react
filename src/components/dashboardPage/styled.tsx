import styled from 'styled-components';

export const DashboardPageContainer = styled.div``;