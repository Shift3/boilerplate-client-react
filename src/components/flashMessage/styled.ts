import styled from 'styled-components';

export const FlashMessageContainer = styled.div``;
