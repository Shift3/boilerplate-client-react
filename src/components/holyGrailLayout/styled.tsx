import styled from 'styled-components';
import { } from '../../utils/styleValues';

export const HolyGrailWrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(6, 1fr);
    grid-auto-rows: minmax(100px, auto);
    grid-template-areas:
      "hd   hd   hd   hd   hd  hd"
      "lsd main main main main rsd"
      "ft   ft   ft   ft   ft  ft";
`;

export const HolyGrailHeader = styled.div`
    grid-area: hd;
`;

export const HolyGrailLeftSideBar = styled.div`
  grid-area: lsd
  `;

export const HolyGrailMain = styled.div`
  grid-area: main
  `;

export const HolyGrailRightSideBar = styled.div`
  grid-area: rsd
  `;

export const HolyGrailFooter = styled.div`
  grid-area: ft
  `;
