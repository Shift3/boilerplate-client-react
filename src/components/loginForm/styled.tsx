import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import styled from 'styled-components';

export const StyledForm = styled(Form)`
  padding: 60px;
  border-radius: 5px;
  min-width: 450px;
`;

export const Title = styled.div`
  color: ${(props) => props.theme.cardHeader};
  font-size: 2em;
  font-style: bold;
  padding-bottom: 10px;
`;

export const FormLabel = styled(Form.Label)`
  color: white;
`;

export const InputError = styled.span`
  color: lightgrey;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding-top: 10px;
`;

export const CancelButton = styled(Button)`
  color: white;
  background-color: ${(props) => props.theme.primary};
  border: 1px solid #679daa;
  margin-right: 20px;
`;

export const LogInButton = styled(Button)`
  color: white;
  background-color: ${(props) => props.theme.accent};
  border: none;
`;
