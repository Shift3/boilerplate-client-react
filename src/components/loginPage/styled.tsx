import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const Wrapper = styled.div`
  height: 100%;
  width: 100%;
  background-color: ${(props) => props.theme.authBackground};
  padding: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const FormLeft = styled.div``;

export const FormRight = styled.div``;

export const CreateAccountButton = styled(Button)``;
