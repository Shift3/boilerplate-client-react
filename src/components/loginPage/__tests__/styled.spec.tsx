import { LoginPageContainer, LoginFormContainer, LoginFormContainerLeft, LoginFormContainerRight } from '../styled';
import { expectToMatchSnapshot } from '../../../utils/test';
import 'jest-styled-components';

describe('Login Page Styled Components', () => {
  it('Should match the stored <LoginPageContainer /> snapshot', () => expectToMatchSnapshot(<LoginPageContainer />));
  it('Should match the stored <LoginFormContainer /> snapshot', () => expectToMatchSnapshot(<LoginFormContainer />));
  it('Should match the stored <LoginFormContainerLeft /> snapshot', () => expectToMatchSnapshot(<LoginFormContainerLeft />));
  it('Should match the stored <LoginFormContainerRight /> snapshot', () => expectToMatchSnapshot(<LoginFormContainerRight />));
});