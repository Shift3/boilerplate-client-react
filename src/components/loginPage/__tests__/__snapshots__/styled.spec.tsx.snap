// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogInPage /> styled components Should match the stored <CreateAccountButton/> snapshot 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  type="button"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <LeftLogin/> snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

<div
  className="c0"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <LinkWrapper/> snapshot 1`] = `
.c0 {
  padding-bottom: 30px;
  font-size: 1.2em;
}

<div
  className="c0"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <LoginWrapper/> snapshot 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50vw;
  height: 50vh;
  padding: 40px;
  margin-left: 100px;
  border-radius: 5px;
}

<div
  className="c0"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <RightLogin/> snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 50px;
  max-width: 350px;
  min-height: 400px;
}

<div
  className="c0"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <Text/> snapshot 1`] = `
.c0 {
  color: white;
}

<div
  className="c0"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <Title/> snapshot 1`] = `
.c0 {
  font-style: bold;
  padding-bottom: 10px;
  width: 200px;
}

<div
  className="c0"
/>
`;

exports[`<LogInPage /> styled components Should match the stored <Wrapper/> snapshot 1`] = `
.c0 {
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;
